rulesets/src/naming.ruleset.yml

```diff
@@ -52,19 +94,162 @@ rules:
         functionOptions:
           match: "^string$"
 
-  # fingerprint property usage and naming
-  sps-fingerprint-naming:
-    description: Rather than property names refering to the implementation for 'hash' or 'hashkey', you MUST use the property name 'fingerprint'.
-    message: "{{property}} is not using property name fingerprint."
+  ##### Standard Properties #####
+  sps-invalid-id-type:
+    description: id SHOULD use a data type of 'string'.
+    severity: warn
+    formats: [oas3]
+    given: '$..[?(@property === "id")].type'
+    then:
+      function: pattern
+      functionOptions:
+        match: "^string$"
+  
+  sps-mandate-abbreviations-identifier:
+    description: Use abbreviations instead of long form names, i.e. identifier SHOULD BE id.
+    severity: warn
+    formats: [oas3]
+    given: '$..properties.*~'
+    then:
+      function: pattern
+      functionOptions:
+        notMatch: "(^identifier([A-Z]|$)|.*Identifier([A-Z]|$))"
+
+  sps-mandate-abbreviations-reference:
+    description: Use abbreviations instead of long form names, i.e. reference SHOULD BE ref.
+    severity: warn
+    formats: [oas3]
+    given: '$..properties.*~'
+    then:
+      function: pattern
+      functionOptions:
+        notMatch: "(^reference([A-Z]|$)|Reference([A-Z]|$))"
+
+  sps-mandate-abbreviations-organization:
+    description: Use abbreviations instead of long form names, i.e. organization SHOULD BE org.
+    severity: warn
+    formats: [oas3]
+    given: '$..properties.*~'
+    then:
+      function: pattern
+      functionOptions:
+        notMatch: "(^organization([A-Z]|$)|Organization([A-Z]|$))"
+
+  sps-invalid-ref-type:
+    description: ref MUST use a data type of 'string'.
     severity: error
     formats: [oas3]
-    given: "$.components.schemas..properties.*~"
+    given: '$..[?(@property === "ref")].type'
     then:
       function: pattern
       functionOptions:
-        notMatch: "^hashkey|hashKey|hash$"
+        match: "^string$"
+
+  sps-invalid-orgid-type:
+    description: orgId MUST use a data type of 'string'.
+    severity: error
+    formats: [oas3]
+    given: '$..[?(@property === "orgId")].type'
+    then:
+      function: pattern
+      functionOptions:
+        match: "^string$"
+
+  sps-invalid-name-type:
+    description: name MUST use a data type of 'string'.
+    severity: error
+    formats: [oas3]
+    given: '$..[?(@property === "name")].type'
+    then:
+      function: pattern
+      functionOptions:
+        match: "^string$"
+
+  sps-invalid-description-type:
+    description: description MUST use a data type of 'string'.
+    severity: error
+    formats: [oas3]
+    given: '$..[?(@property === "description")].type'
+    then:
+      function: pattern
+      functionOptions:
+        match: "^string$"
+
+  sps-invalid-requestid-type:
```

Comment:
I think we should stick to strict kebab case for these: `sps-invalid-request-id-type` and `sps-invalid-created-date-time-type` etc